### PR TITLE
feat(shared-create-profile-form): [skp-111] created profile form

### PR DESCRIPTION
**Pull request by Neutronas**

### CODE REVIEW

File: libs/shared/profile-form/src/lib/shared-profile-form.tsx [hunk 1]
```diff
@@ -0,0 +1,258 @@
+import React from 'react';
```
Comment (doncevas) [inline libs/shared/profile-form/src/lib/shared-profile-form.tsx:1]: use import React, {useState} from 'react';

File: libs/shared/profile-form/src/lib/shared-profile-form.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  }) => void;
+  categories: data[];
+  cities: data[];
+  countries: data[];
+}
+
+export const SharedProfileForm = (props: SharedProfileFormProps) => {
+  const { handleSubmit, register, control, errors } = useForm();
+
+  const [categories, setCategories] = React.useState<string[]>([]);
```
Comment (doncevas) [inline libs/shared/profile-form/src/lib/shared-profile-form.tsx:44]: remove React. Use only useState<string[]>([]) same below

File: libs/shared/profile-form/src/lib/shared-profile-form.tsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  const [city, setCity] = React.useState<string>('');
+  const [country, setCountry] = React.useState<string>('');
+
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setCategories(event.target.value as string[]);
```
Comment (doncevas) [inline libs/shared/profile-form/src/lib/shared-profile-form.tsx:49]: don't use 'as'

File: libs/shared/profile-form/src/lib/shared-profile-form.tsx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+export const SharedProfileForm = (props: SharedProfileFormProps) => {
+  const { handleSubmit, register, control, errors } = useForm();
+
+  const [categories, setCategories] = React.useState<string[]>([]);
+  const [city, setCity] = React.useState<string>('');
+  const [country, setCountry] = React.useState<string>('');
+
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
```
Comment (doncevas) [inline libs/shared/profile-form/src/lib/shared-profile-form.tsx:48]: don't use unknown and allso remove React. and ChangeEvent import same as useState

File: libs/shared/profile-form/src/lib/shared-profile-form.tsx [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+              onChange={([selected]) => {
+                return selected;
+              }}
+              as={
+                <Select
+                  labelId="city-label"
+                  id="city"
+                  name="city"
+                  value={city}
+                  onChange={(event) => setCity(event.target.value as string)}
```
Comment (doncevas) [inline libs/shared/profile-form/src/lib/shared-profile-form.tsx:183]: don't use 'as'

File: libs/shared/profile-form/src/lib/shared-profile-form.tsx [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+              onChange={([selected]) => {
+                return selected;
+              }}
+              as={
+                <Select
+                  labelId="country-label"
+                  id="country"
+                  name="country"
+                  value={country}
+                  onChange={(event) => setCountry(event.target.value as string)}
```
Comment (doncevas) [inline libs/shared/profile-form/src/lib/shared-profile-form.tsx:214]: don't use 'as'

File: libs/shared/profile-form/src/lib/shared-profile-form.tsx (hunk 5 above)
Comment (Neutronas) [reply libs/shared/profile-form/src/lib/shared-profile-form.tsx:183]: Removing it gives me an error
`Argument of type 'unknown' is not assignable to parameter of type 'SetStateAction<string>'.
  Type 'unknown' is not assignable to type '(prevState: string) => string'.
    Type '{}' provides no match for the signature '(prevState: string): string'.ts(2345)`

**Pull request merged**